### PR TITLE
skip CMake 3.30.0

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/80

Adds constraints to avoid pulling in CMake 3.30.0, for the reasons described in that issue.

### CODE REVIEW

File: ci/wheel_smoke_test_cugraph_kg.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 import cudf
 import cugraph
-import cugraph-kg
+import cugraph_pg
```
Comment (jameslamb) [inline ci/wheel_smoke_test_cugraph_kg.py:16]: I wasn't sure what this should be, as I see references to both `cugraph_pg`

https://github.com/rapidsai/cugraph-pg/blob/d0d101a53c89984841199c3555bde6ee5d8658be/python/cugraph-pg/pyproject.toml#L16

and `cugraph_kg`

https://github.com/rapidsai/cugraph-pg/blob/d0d101a53c89984841199c3555bde6ee5d8658be/conda/recipes/cugraph/meta.yaml#L11

Comment (bdice) [reply ci/wheel_smoke_test_cugraph_kg.py:16]: I don't even seen this import being used in the smoke test. I would ignore this for now if you can. Admin-merge or something like that. This package isn't even named in a legal way (it uses dashes in places that aren't allowed) so I would put this back on library devs to resolve first.

File: dependencies.yaml [hunk 2]
```diff
@@ -70,7 +70,7 @@ dependencies:
       - output_types: conda
         packages:
           - c-compiler
-          - cmake>=3.26.4
+          - &cmake_ver cmake>=3.26.4
```
Comment (bdice) [inline dependencies.yaml:73]:
```suggestion
          - &cmake_ver cmake>=3.26.4,!=3.30.0
```

Comment (jameslamb) [reply dependencies.yaml]: blegh thank you (I just pushed this change as part of https://github.com/rapidsai/cugraph-pg/pull/23/commits/0dd89ed42e8ad2fcfa202c51f09d857ca298416e)

File: ci/wheel_smoke_test_cugraph_kg.py (hunk 1 above)
Comment (jameslamb) [reply ci/wheel_smoke_test_cugraph_kg.py]: Alright sure, I just reverted the changes to this file. It won't require an admin-merge, as CI isn't set up here yet.

**Review by bdice — APPROVED** [review]

**Comment by jakirkham** [comment]
Discussed offline with @BradReesWork , this will be picked up before cuGraph-PG does its first release

**Pull request merged**